unix/lofs9p.ml

```diff
@@ -352,7 +352,11 @@ module New(Params : sig val root : string list end) = struct
     | exception Not_found -> bad_fid
     | path ->
       let realpath = Path.realpath path in
-      Lwt_unix.unlink realpath
+      Printf.fprintf stderr "Calling unlink on %s\n%!" realpath;
+      Lwt_unix.LargeFile.stat realpath
+      >>= fun stats ->
+      ( if stats.Lwt_unix.LargeFile.st_kind = Lwt_unix.S_DIR
+        then Lwt_unix.rmdir else Lwt_unix.unlink ) realpath
```

Comment:
Slightly confusingly, an `unlink` of a directory will fail with `EPERM` if the user isn't root. I'll ignore that for now :/